### ISSUE TITLE
[Feature request]: Real-time processing

### ISSUE THREAD

**Issue by andrewferguson** [markmerz/ladspa-sq-sv#1]
I've compiled and setup the plugin (and it's working really well - a fantastic job!), and I'd love to be able to use a Raspberry Pi with an external audio card (the one I have is the Wolfson Audio card) as well as the Pi's built in audio output, to link it to my amplifier so that it could receive the output of my record player through the Wolfson Audio card's line-level input, process it using your plugin, then output the front left/right channels to the Wolfson Audio card's line-level output, and the rear left/right channels to the Pi's in-built audio output.

The issue is I've never used ecasound before, so I don't know if piping in audio from a hardware source, then piping various output channels to different hardware sources is something that can be done within ecasound, or if the plugins themselves need to be altered.

Latency is not an issue - I don't mind if it takes a second or so for the Pi to process the incoming signal and output it back into the amplifier.

**Comment by andrewferguson** [comment]
So I tried
`ecasound -a:1,2 -x -i sq-test-tones.m4a -el:hilbert_stereo_200taps -el:sq_decoder_shadow_vector -a:1 -f:16,2,44100 -o front.wav -a:2 -f:16,2,44100 -o back.wav -chcopy:3,1 -chcopy:4,2`
to split the output into two files, one for the front and one for the back, and although this worked individually (either outputting for the front or for the back) I can't seem to get this to work when doing both together.

**Comment by markmerz** [comment]
Using multiple sound cards in parallel is very special can of worms but its out of scope of this repo. Some hints:
* Tact crystals run in slightly different speeds and one of the cards will have buffer underrun soon enough.
* Pulseaudio has dedicated code for seamlessly syncing multiple sound cards by either changing clock speeds on cards which support that or by resampling streams. https://www.freedesktop.org/wiki/Software/PulseAudio/FAQ/#index32h3 has examples.
